### PR TITLE
deps: gateway API bumped to v1.0.0

### LINKED ISSUE #96: Bump Gateway API to v1.0.0
**What would you like to be added**:

Now that The Gateway API is GA and `v1.0.0` is [out](https://github.com/kubernetes-sigs/gateway-api/releases/tag/v1.0.0), we should bump the Gateway API dependencies to v1.0.0 as well.

### PR DESCRIPTION
**Pull request by mlavacca**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #96 

**Does this PR introduce a user-facing change?**:

```release-note
Gateway API has been bumped to v1.0.0.
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/kubernetes-sigs/ingress2gateway
 
-go 1.20
+go 1.21
+
+toolchain go1.21.1
```
Comment (LiorLieberman) [inline go.mod:5]: not sure we wants this line.
image

**Review by LiorLieberman — COMMENTED** [review]
one small nit.

File: go.mod (hunk 1 above)
Comment (mlavacca) [reply go.mod:5]: It was automatically added after bumping to 1.21. Removed 👍

**Pull request merged**